frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/mention/child_page_transaction_handler.dart

```diff
@@ -0,0 +1,228 @@
+          mention,
+          isPaste,
+          parentViewId,
+          isCut,
+        );
+      }
+
+      if (context.mounted) {
+        context.read<ClipboardState>().endHandlingPaste(_pasteIdentifier);
+      }
+    }
+  }
+
+  Future<void> _handleDeletion(
+    BuildContext context,
+    MentionBlockData data,
+  ) async {
+    final viewId = data.$2[MentionBlockKeys.pageId];
+
+    final result = await ViewBackendService.deleteView(viewId: viewId);
+    result.fold(
+      (_) {},
+      (error) {
+        Log.error(error);
+        if (context.mounted) {
+          showSnapBar(
+            context,
+            LocaleKeys.document_plugins_subPage_errors_failedDeletePage.tr(),
+          );
+        }
+      },
+    );
+  }
+
+  Future<void> _handleAddition(
+    BuildContext context,
+    EditorState editorState,
+    MentionBlockData data,
+    bool isPaste,
+    String? parentViewId,
+    bool isCut,
+  ) async {
+    if (parentViewId == null) {
+      return;
+    }
+
+    final viewId = data.$2[MentionBlockKeys.pageId];
+    if (isPaste && !isCut) {
+      _handlePasteFromCopy(
+        context,
+        editorState,
+        data.$1,
+        data.$3,
+        viewId,
+        parentViewId,
+      );
+    } else {
+      _handlePasteFromCut(viewId, parentViewId);
+    }
+  }
+
+  void _handlePasteFromCut(String viewId, String parentViewId) async {
+    // Attempt to restore from Trash just in case
+    await TrashService.putback(viewId);
+
+    final view = (await ViewBackendService.getView(viewId)).toNullable();
+    if (view == null) {
+      return Log.error('View not found: $viewId');
+    }
+
+    if (view.parentViewId == parentViewId) {
+      return;
+    }
+
+    await ViewBackendService.moveViewV2(
+      viewId: viewId,
+      newParentId: parentViewId,
+      prevViewId: null,
+    );
+  }
+
+  void _handlePasteFromCopy(
+    BuildContext context,
+    EditorState editorState,
+    Node node,
+    int index,
+    String viewId,
+    String parentViewId,
+  ) async {
+    final view = (await ViewBackendService.getView(viewId)).toNullable();
+    if (view == null) {
+      return Log.error('View not found: $viewId');
+    }
+
+    final duplicatedViewOrFailure = await ViewBackendService.duplicate(
+      view: view,
+      openAfterDuplicate: false,
+      includeChildren: true,
+      syncAfterDuplicate: true,
+      parentViewId: parentViewId,
+    );
+
+    await duplicatedViewOrFailure.fold(
+      (newView) async {
+        final newMentionAttributes = {
+          MentionBlockKeys.mention: {
+            MentionBlockKeys.type: MentionType.childPage.name,
+            MentionBlockKeys.pageId: newView.id,
+          },
+        };
+
+        // The index is the index of the delta, to get the index of the mention character
+        // in all the text, we need to calculate it based on the deltas before the current delta.
+        int mentionIndex = 0;
+        for (final (i, delta) in node.delta!.indexed) {
```

Comment:
use `?`

Reply:
It can't reach here without the mention existing in the node delta, so it cannot be null, I can change it, but we are guaranteed that it is not null here.